Common/tests/DependencyResolverTest.cs

```diff
@@ -0,0 +1,587 @@
+                                               },
+                                               new Output(false,
+                                                          "")),
+                                  new TaskData("SessionId",
+                                               TaskCancelling,
+                                               "",
+                                               "",
+                                               "PayloadId",
+                                               new[]
+                                               {
+                                                 "parent1",
+                                               },
+                                               Array.Empty<string>(),
+                                               new[]
+                                               {
+                                                 "output1",
+                                               },
+                                               Array.Empty<string>(),
+                                               TaskStatus.Cancelling,
+                                               options with
+                                               {
+                                                 PartitionId = "part2",
+                                               },
+                                               new Output(false,
+                                                          "")),
+                                  new TaskData("SessionId",
+                                               "TaskFailedId",
+                                               "OwnerPodId",
+                                               "OwnerPodName",
+                                               "PayloadId",
+                                               new[]
+                                               {
+                                                 "parent1",
+                                               },
+                                               new[]
+                                               {
+                                                 CreatedData1,
+                                               },
+                                               new[]
+                                               {
+                                                 "output1",
+                                               },
+                                               Array.Empty<string>(),
+                                               TaskStatus.Error,
+                                               options,
+                                               new Output(false,
+                                                          "sad task")),
+                                })
+                   .ConfigureAwait(false);
+
+    return provider;
+  }
+
+
+  [Test]
+  public async Task Init()
```

Comment:
This looks more like your `[SetUp]`

Reply:
I did not want to struggle with the provider that should be nullable in this case